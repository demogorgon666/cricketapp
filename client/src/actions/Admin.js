import {
  GET_PLAYERS,
  CREATE_PLAYER,
  CREATE_TEAM,
  EDIT_PLAYER,
  EDIT_TEAM,
  DELETE_PLAYER,
  DELETE_TEAM,
  GET_PLAYER_SEARCH,
  GET_TEAM_SEARCH,
  GET_ALLTEAMS,
  ERROR_TYPE
} from "./Types";
import axios from "axios";

// export const getPlayers = () => dispatch => {
// 	return axios
// 		.get("http://localhost:5000/api/cricketalpha/allPlayer")
// 		.then(res => {
// 			console.log("From Player Actions", res.data.data);
// 			dispatch({
// 				type: GET_PLAYERS,
// 				payload: res.data.data
// 			});
// 		})
// 		.catch(err => {
// 			console.log(err);
// 		});
// };
//
export const getPlayers = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

// export const getTeams = () => dispatch => {
//     return axios
//       .get("http://localhost:5000/api/cricketalpha/allteams")
//       .then(res => {
//         dispatch({
//           type: GET_TEAMS,
//           payload: res.data.data
//         });
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

export const getAllTeams = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/admin/allteam", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_ALLTEAMS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

//   export const createPlayer = player => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/player/new", player)
//       .then(res => {
//         dispatch({
//           type: CREATE_PLAYER
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

//   export const createTeam = team => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/user/new", team)
//       .then(res => {
//         dispatch({
//           type: CREATE_TEAM
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };
export const createTeam = team => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/team/new", team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_TEAM
      });
      alert("Added one team successfully");
      dispatch(getAllTeams());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const createPlayer = player => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/player/new", player, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_PLAYER
      });
      alert("Added one player successfully");
      dispatch(getPlayers());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const editPlayer = player => dispatch => {
  return axios
    .put(
      "http://localhost:5000/apis/admin/editplayer/" + player.player_id,
      player,
      {
        headers: {
          Authorization: `Bearer ${localStorage.getItem("token")}`
        }
      }
    )
    .then(res => {
      dispatch({
        type: EDIT_PLAYER
      });

      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const editTeam = team => dispatch => {
  return axios
    .put("http://localhost:5000/apis/admin/editteam/" + team.team_id, team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: EDIT_TEAM
      });
      // dispatch(getUsers());
      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deletePlayer = player_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteplayer/" + player_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_PLAYER
      });
      dispatch(getPlayers());
      // alert("Player Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deleteTeam = team_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteteam/" + team_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_TEAM
      });
      dispatch(getAllTeams());
      // alert("Team Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const getPlayerSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/search/" + searchString)
    .then(res => {
      dispatch({
        type: GET_PLAYER_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTeamSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/searchteam/" + searchString)
    .then(res => {
      dispatch({
        type: GET_TEAM_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
