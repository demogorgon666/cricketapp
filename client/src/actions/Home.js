import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
