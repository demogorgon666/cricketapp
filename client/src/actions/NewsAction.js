import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
