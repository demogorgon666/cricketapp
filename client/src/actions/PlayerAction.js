import {
  GET_PLAYERS,
  GET_SINGLE_PLAYER,
  GET_ALL_BATSMAN,
  GET_ALL_BOWLERS,
  GET_BATSMAN_STATS,
  GET_ODI_BATSMAN_STATS,
  GET_T20_BATSMAN_STATS,
  GET_ODI_BOWLER_STATS,
  GET_T20_BOWLER_STATS,
  GET_TEST_BOWLER_STATS,
  GET_TOP_SIXES
} from "./Types";
import axios from "axios";

export const getPlayers = () => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer")
    .then(res => {
      console.log("From Player Actions", res.data.data);
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getSinglePlayer = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/singlePlayer/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions Single player is: ",
        res.data.data[0].player_name
      );
      dispatch({
        type: GET_SINGLE_PLAYER,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmen = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBatsman", match_type)
    .then(res => {
      // console.log("From Player Actions batsman data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BATSMAN,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBowlers = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBowlers", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BOWLERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTopSixes = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopSixes", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_TOP_SIXES,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODI-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions odi Batsman stats is: ", res.data.ODI);
      dispatch({
        type: GET_ODI_BATSMAN_STATS,
        payload: res.data.ODI
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODIBowlerStats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions odi BOWLER stats is: ",
        res.data.ODIBowler
      );
      dispatch({
        type: GET_ODI_BOWLER_STATS,
        payload: res.data.ODIBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBatsmanStats = player_id => dispatch => {
  return axios
    .get(
      "http://localhost:5000/apis/PlayerInfo/Test-Batsman-Stats/" + player_id
    )
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.Test);
      dispatch({
        type: GET_BATSMAN_STATS,
        payload: res.data.Test
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Test-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions test Batsman stats is: ",
        res.data.TestBowler
      );
      dispatch({
        type: GET_TEST_BOWLER_STATS,
        payload: res.data.TestBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.T20);
      dispatch({
        type: GET_T20_BATSMAN_STATS,
        payload: res.data.T20
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions ODI BOWLER stats is: ",
        res.data.T20Bowler
      );
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.T20Bowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Batsman-Stats/" + player_id)
    .then(res => {
      console.log("From Player Actions test Batsman stats is: ", res.data.data);
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
