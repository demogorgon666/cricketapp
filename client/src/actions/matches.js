import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

