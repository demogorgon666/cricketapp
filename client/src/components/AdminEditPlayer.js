import React, { Component } from "react";
import { Link } from "react-router-dom";
import { getPlayers, editPlayer } from "../actions/Admin";
import { connect } from "react-redux";
import "./css/AdminEditPage.css";
// import NavBar from "./common/Navbar";

import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

export class AdminEditPlayer extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);

    // this.state = {
    //   player_gender: "Gender",
    //   batting_style: "Batting Style",
    //   bowling_style: "Bowling style",
    //   player_role: "Player Role"
    // };
    this.OnSelectGender = this.OnSelectGender.bind(this);
    this.OnSelectBattingStyle = this.OnSelectBattingStyle.bind(this);
    this.OnSelectBowlerStyle = this.OnSelectBowlerStyle.bind(this);
    this.OnSelectRole = this.OnSelectRole.bind(this);
    this.OnChange = this.OnChange.bind(this);
    this.OnEdit = this.OnEdit.bind(this);
  }
  OnSelectGender(e) {
    this.setState({
      player_gender: e.target.value
    });
  }
  OnSelectBattingStyle(e) {
    this.setState({
      batting_style: e.target.value
    });
  }
  OnSelectBowlerStyle(e) {
    this.setState({
      bowling_style: e.target.value
    });
  }
  OnSelectRole(e) {
    this.setState({
      player_role: e.target.value
    });
  }

  state = {
    player_id: this.props.match.params.player_id,
    player_name: this.props.history.location.state.players.player_name,
    player_country: this.props.history.location.state.players.player_country,
    batting_style: this.props.history.location.state.players.batting_style,
    bowling_style: this.props.history.location.state.players.bowling_style,
    player_gender: this.props.history.location.state.players.player_gender,
    player_role: this.props.history.location.state.players.player_role,
    debut_odi_match: this.props.history.location.state.players.debut_odi_match,
    debut_test_match: this.props.history.location.state.players
      .debut_test_match,
    debut_t20_match: this.props.history.location.state.players.debut_t20_match,
    player_dob: new Date(),
    showError: false,
    errorMessage: ""
  };

  componentDidMount() {
    this.props.getPlayers(this.props.match.params.player_id);
    if (!localStorage.getItem("token")) {
      this.props.history.push("/");
    }
  }
  OnChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleChange = date => {
    this.setState({
      player_dob: date
    });
  };

  OnEdit = e => {
    console.log("hello");
    e.preventDefault();
    if (
      !this.state.player_name ||
      !this.state.player_country ||
      !this.state.player_gender ||
      !this.state.batting_style ||
      !this.state.bowling_style ||
      !this.state.player_dob ||
      !this.state.player_role ||
      !this.state.debut_odi_match ||
      !this.state.debut_test_match ||
      !this.state.debut_t20_match
    ) {
      this.setState({
        showError: true,
        errorMessage: "Enter all the fields"
      });
    } else {
      //   e.preventDefault();

      e.preventDefault();
      if (this.state.player_name.length < 2) {
        this.setState({
          showError: true,
          errorMessage: "Name length should be greater then 1!"
        });
      } else {
        if (this.state.player_gender.length < 2) {
          this.setState({
            showError: true,
            errorMessage: "Please enter valid player type!"
          });
        } else {
          if (this.state.player_country.length < 2) {
            this.setState({
              showError: true,
              errorMessage: "Please enter valid player type!"
            });
          } else {
            if (this.state.batting_style.length < 5) {
              this.setState({
                showError: true,
                errorMessage: "Please enter valid batting style!"
              });
            } else {
              if (this.state.bowling_style.length < 5) {
                this.setState({
                  showError: true,
                  errorMessage: "Please enter valid bowling style!"
                });
              } else {
                if (this.state.player_dob.length < 5) {
                  this.setState({
                    showError: true,
                    errorMessage: "Please enter valid player dob!"
                  });
                } else {
                  if (this.state.player_role.length < 5) {
                    this.setState({
                      showError: true,
                      errorMessage: "Please enter valid player role!"
                    });
                  } else {
                    if (this.state.debut_odi_match.length < 5) {
                      this.setState({
                        showError: true,
                        errorMessage: "Please enter valid debut odi match!"
                      });
                    } else {
                      if (this.state.debut_test_match.length < 5) {
                        this.setState({
                          showError: true,
                          errorMessage: "Please enter valid debut test match!"
                        });
                      } else {
                        if (this.state.debut_t20_match.length < 5) {
                          this.setState({
                            showError: true,
                            errorMessage: "Please enter valid debut t20 match!"
                          });
                        } else {
                          console.log("hello");
                          this.setState({
                            showError: false,
                            errorMessage: ""
                          });
                          let player = {
                            player_id: this.state.player_id,
                            player_name: this.state.player_name,
                            player_country: this.state.player_country,
                            batting_style: this.state.batting_style,
                            bowling_style: this.state.bowling_style,
                            player_dob: this.state.player_dob.toLocaleDateString(
                              "en-GB"
                            ),
                            player_role: this.state.player_role,
                            player_gender: this.state.player_gender,
                            debut_odi_match: this.state.debut_odi_match,
                            debut_test_match: this.state.debut_test_match,
                            debut_t20_match: this.state.debut_t20_match
                          };
                          this.props.editPlayer(player);
                          this.props.history.push("/adminplayer");

                          this.setState({
                            player_name: "",
                            player_country: "",
                            batting_style: "",
                            bowling_style: "",
                            player_dob: "",
                            player_gender: "",
                            player_role: "",
                            debut_odi_match: "",
                            debut_test_match: "",
                            debut_t20_match: ""
                          });
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  };

  render() {
    return (
      <div>
        {/* <NavBar /> */}

        <div id="AdminEditform">
          <fieldset>
            <h1>Edit New Player</h1>
            <input
              type="text"
              name="player_name"
              placeholder="Player name"
              value={this.state.player_name}
              onChange={this.OnChange}
            />
            <input
              type="text"
              name="player_country"
              placeholder="Enter Player Country"
              value={this.state.player_country}
              onChange={this.OnChange}
            />
            <div className="playing-style">
              <div>
                <select
                  type="text"
                  name="batting_style"
                  value={this.state.batting_style}
                  onChange={this.OnSelectBattingStyle}
                  className="choice"
                >
                  <option name="choice">Batting style</option>
                  <option name="male">Right Hand</option>
                  <option name="male">left Hand</option>
                </select>
              </div>
              <div>
                <select
                  type="text"
                  name="bowling_style"
                  value={this.state.bowling_style}
                  onChange={this.OnSelectBowlerStyle}
                  className="choice"
                >
                  <option name="choice">Bowling style</option>
                  <option name="male">Right-arm fast</option>
                  <option name="male">Right-arm fast medium</option>
                  <option name="male">Right-arm medium fast</option>
                  <option name="male">Right-arm medium</option>
                  <option name="male">Left-arm fast</option>
                  <option name="male">Left-arm fast medium</option>
                  <option name="male">Left-arm medium fast</option>
                  <option name="male">Left-arm medium</option>
                  <option name="male">Off break (right-arm)</option>
                  <option name="male">Leg break (right-arm)</option>
                  <option name="male">Slow left-arm orthodox</option>
                  <option name="male">Slow left-arm chinaman</option>
                </select>
              </div>
            </div>
            <div className="playing-style">
              <DatePicker
                name="datepicker"
                dateFormat="yyyy-MM-dd"
                placeholderText="DOB"
                selected={this.state.player_dob}
                onChange={this.handleChange}
              />

              <div>
                <select
                  type="text"
                  name="player_gender"
                  value={this.state.player_gender}
                  onChange={this.OnSelectGender}
                  className="choice"
                >
                  <option name="choice">Gender</option>
                  <option name="male">male</option>
                  <option name="male">female</option>
                </select>
              </div>
            </div>
            <select
              type="text"
              name="player_role"
              placeholder="Enter Player Role"
              value={this.state.player_role}
              onChange={this.OnSelectRole}
              className="choice"
            >
              <option name="choice">Player Role</option>
              <option name="male">Wicket-keeper-batsmen</option>
              <option name="male">Wicket-keeper-batswomen</option>
              <option name="male">Bowler</option>
              <option name="male">Batsmen</option>
              <option name="male">Batswomen</option>
              <option name="male">Allrounder</option>
            </select>
            <input
              type="text"
              name="debut_odi_match"
              placeholder="Enter Debut ODI Match"
              value={this.state.debut_odi_match}
              onChange={this.OnChange}
            />
            <input
              type="text"
              name="debut_test_match"
              placeholder="Enter Debut Test Match"
              value={this.state.debut_test_match}
              onChange={this.OnChange}
            />
            <input
              type="text"
              name="debut_t20_match"
              placeholder="Enter Debut T20 Match"
              value={this.state.debut_t20_match}
              onChange={this.OnChange}
            />
            <span
              className="errorMessage"
              style={{
                color: "#c0392b",
                display: this.state.showError ? "block" : "none"
              }}
            >
              {this.state.errorMessage}
            </span>
            <div className="form-button">
              <Link to="/adminplayer">
                <button className="cancel">Cancel</button>
              </Link>

              <button
                className="admineditpagebutton"
                onChange={this.OnChange}
                onClick={this.OnEdit}
              >
                Edit Player
                {console.log(this.OnEdit)}
              </button>
            </div>
          </fieldset>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  player: state.AdminPlayerReducer.player,
  playerInfo: state.PlayerReducer.playerInfo
});

export default connect(
  mapStateToProps,
  { getPlayers, editPlayer }
)(AdminEditPlayer);
