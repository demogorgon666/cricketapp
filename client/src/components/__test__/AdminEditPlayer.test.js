import React from "react";
import { shallow } from "enzyme";
import { AdminEditPlayer } from "../AdminEditPlayer";
// import store from "../../store";
// import { Provider } from "react-redux";
const admineditplayer = jest.fn();
const state = jest.fn();
import { getPlayers } from "../../actions/Admin";

const wrapper = shallow(
  <AdminEditPlayer
    getPlayers={getPlayers}
    admineditplayer={admineditplayer}
    match={{ isExact: true, params: { path: "/", url: "/" } }}
    history={{
      location: {
        state: {
          players: {
            player_name: "test",
            player_country: "test",
            batting_style: "test",
            bowling_style: "test",
            player_gender: "male",
            player_role: "batsmen",
            debut_odi_match: "pakistan",
            debut_test_match: "pakistan",
            debut_t20_match: "paksitan"
          }
        }
      }
    }}
  />
);

describe("test  Component", () => {
  it("render the component", () => {
    // const wrapper = shallow(<Register register={register} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("there should be a div", () => {
    expect(wrapper.find("div").length).toBe(8);
  });
  // it("there should be a navbar", () => {
  //   expect(wrapper.find("NavBar").length).toBe(1);
  // });
  it("there should be a h1", () => {
    expect(wrapper.find("h1").length).toBe(1);
    expect(
      wrapper
        .find("h1")
        .at(0)
        .text()
    ).toBe("Edit New Player");
  });
  it("there should be a input", () => {
    expect(wrapper.find("input").length).toBe(5);
  });
  it("there should be a buttons", () => {
    expect(wrapper.find("button").length).toBe(2);
  });
  // it("there should be a link", () => {
  //   expect(wrapper.find("Link").length).toBe(2);
  // });

  it("there should be a fieldset tag", () => {
    expect(wrapper.find("fieldset").length).toBe(1);
  });
  it("there should be a select tag", () => {
    expect(wrapper.find("select").length).toBe(4);
  });
  // it("there should be a p tag", () => {
  //   expect(wrapper.find("p").length).toBe(1);
  //   expect(wrapper.find("p").text()).toBe("Already have an account ?Login");
  // });
  it("there should be a button", () => {
    expect(
      wrapper
        .find("button")
        .at(0)
        .text()
    ).toBe("Cancel");
    expect(
      wrapper
        .find("button")
        .at(1)
        .text()
    ).toBe("Edit Player");
  });

  // it("should change the state of testClicked to true when test tab is clicked", () => {
  //   const test = wrapper.find("button").at(0);
  //   test.simulate("click");
  //   expect(wrapper.state().onRegister).toBe(true);
  // });

  it("there should be a span", () => {
    expect(wrapper.find("span").length).toBe(1);
  });

  it("there should type inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(2)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(3)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(4)
        .prop("type")
    ).toEqual("text");
  });

  it("there should name in  inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("name")
    ).toEqual("player_name");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("name")
    ).toEqual("player_country");

    expect(
      wrapper
        .find("input")
        .at(2)
        .prop("name")
    ).toEqual("debut_odi_match");
    expect(
      wrapper
        .find("input")
        .at(3)
        .prop("name")
    ).toEqual("debut_test_match");
    expect(
      wrapper
        .find("input")
        .at(4)
        .prop("name")
    ).toEqual("debut_t20_match");
  });
  it("there should name in  inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("placeholder")
    ).toEqual("Player name");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("placeholder")
    ).toEqual("Enter Player Country");

    expect(
      wrapper
        .find("input")
        .at(2)
        .prop("placeholder")
    ).toEqual("Enter Debut ODI Match");
    expect(
      wrapper
        .find("input")
        .at(3)
        .prop("placeholder")
    ).toEqual("Enter Debut Test Match");
    expect(
      wrapper
        .find("input")
        .at(4)
        .prop("placeholder")
    ).toEqual("Enter Debut T20 Match");
  });
});
