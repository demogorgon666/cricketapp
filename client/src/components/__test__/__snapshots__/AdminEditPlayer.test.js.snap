// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Component render the component 1`] = `
<div>
  <Connect(AdminNavbar) />
  <div
    id="AdminEditform"
  >
    <fieldset>
      <h1
        id="heading"
      >
        Edit New Player
      </h1>
      <input
        id="player_name"
        name="player_name"
        onChange={[Function]}
        placeholder="Player name"
        type="text"
        value="test"
      />
      <input
        id="player_country"
        name="player_country"
        onChange={[Function]}
        placeholder="Enter Player Country"
        type="text"
        value="test"
      />
      <div
        className="playing-style"
      >
        <div>
          <select
            className="choice"
            id="batting_style"
            name="batting_style"
            onChange={[Function]}
            type="text"
            value="test"
          >
            <option
              name="choice"
            >
              Batting style
            </option>
            <option
              name="male"
            >
              Right Hand
            </option>
            <option
              name="male"
            >
              left Hand
            </option>
          </select>
        </div>
        <div>
          <select
            className="choice"
            id="bowling_style"
            name="bowling_style"
            onChange={[Function]}
            type="text"
            value="test"
          >
            <option
              name="choice"
            >
              Bowling style
            </option>
            <option
              name="male"
            >
              Right-arm fast
            </option>
            <option
              name="male"
            >
              Right-arm fast medium
            </option>
            <option
              name="male"
            >
              Right-arm medium fast
            </option>
            <option
              name="male"
            >
              Right-arm medium
            </option>
            <option
              name="male"
            >
              Left-arm fast
            </option>
            <option
              name="male"
            >
              Left-arm fast medium
            </option>
            <option
              name="male"
            >
              Left-arm medium fast
            </option>
            <option
              name="male"
            >
              Left-arm medium
            </option>
            <option
              name="male"
            >
              Off break (right-arm)
            </option>
            <option
              name="male"
            >
              Leg break (right-arm)
            </option>
            <option
              name="male"
            >
              Slow left-arm orthodox
            </option>
            <option
              name="male"
            >
              Slow left-arm chinaman
            </option>
          </select>
        </div>
      </div>
      <div
        className="playing-style"
      >
        <t
          allowSameDay={false}
          dateFormat="yyyy-MM-dd"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          id="datepicker"
          inlineFocusSelectedMonth={false}
          monthsShown={1}
          name="datepicker"
          nextMonthButtonLabel="Next Month"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="DOB"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2019-10-22T06:35:28.445Z}
          shouldCloseOnSelect={true}
          showMonthYearPicker={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={false}
          strictParsing={false}
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
        />
        <div>
          <select
            className="choice"
            id="player_gender"
            name="player_gender"
            onChange={[Function]}
            type="text"
            value="male"
          >
            <option
              name="choice"
            >
              Gender
            </option>
            <option
              name="male"
            >
              male
            </option>
            <option
              name="male"
            >
              female
            </option>
          </select>
        </div>
      </div>
      <select
        className="choice"
        id="player_role"
        name="player_role"
        onChange={[Function]}
        placeholder="Enter Player Role"
        type="text"
        value="batsmen"
      >
        <option
          name="choice"
        >
          Player Role
        </option>
        <option
          name="male"
        >
          Wicket-keeper-batsmen
        </option>
        <option
          name="male"
        >
          Wicket-keeper-batswomen
        </option>
        <option
          name="male"
        >
          Bowler
        </option>
        <option
          name="male"
        >
          Batsmen
        </option>
        <option
          name="male"
        >
          Batswomen
        </option>
        <option
          name="male"
        >
          Allrounder
        </option>
      </select>
      <input
        id="debut_odi_match"
        name="debut_odi_match"
        onChange={[Function]}
        placeholder="Enter Debut ODI Match"
        type="text"
        value="pakistan"
      />
      <input
        id="debut_test_match"
        name="debut_test_match"
        onChange={[Function]}
        placeholder="Enter Debut Test Match"
        type="text"
        value="pakistan"
      />
      <input
        id="debut_t20_match"
        name="debut_t20_match"
        onChange={[Function]}
        placeholder="Enter Debut T20 Match"
        type="text"
        value="paksitan"
      />
      <span
        className="errorMessage"
        style={
          Object {
            "color": "#c0392b",
            "display": "none",
          }
        }
      />
      <div
        className="form-button"
      >
        <Link
          to="/adminplayer"
        >
          <button
            className="cancel"
            id="cancel"
          >
            Cancel
          </button>
        </Link>
        <button
          className="admineditpagebutton"
          id="editbutton"
          onChange={[Function]}
          onClick={[Function]}
        >
          Edit Player
        </button>
      </div>
    </fieldset>
  </div>
</div>
`;
